web/src/settings_users.ts

```diff
@@ -269,39 +331,48 @@ function get_last_active(user) {
     return timerender.render_now(last_active_date).time_str;
 }
 
-function human_info(person) {
-    const info = {};
-
-    info.is_bot = false;
-    info.user_role_text = people.get_user_type(person.user_id);
-    info.is_active = people.is_person_active(person.user_id);
-    info.user_id = person.user_id;
-    info.full_name = person.full_name;
-    info.bot_owner_id = person.bot_owner_id;
-
-    info.can_modify = current_user.is_admin;
-    info.is_current_user = people.is_my_user_id(person.user_id);
-    info.cannot_deactivate =
-        person.is_owner && (!current_user.is_owner || people.is_current_user_only_owner());
-    info.display_email = person.delivery_email;
-    info.img_src = people.small_avatar_url_for_person(person);
-
-    // TODO: This is not shown in deactivated users table and it is
-    // controlled by `display_last_active_column` We might just want
-    // to show this for deactivated users, too, even though it might
-    // usually just be undefined.
-    info.last_active_date = get_last_active(person);
-
-    return info;
+function human_info(person: User): {
+    is_bot: false;
+    user_role_text: string | undefined;
+    is_active: boolean;
+    user_id: number;
+    full_name: string;
+    bot_owner_id: number | null;
+    can_modify: boolean;
+    is_current_user: boolean;
+    cannot_deactivate: boolean;
+    display_email: string | null;
+    img_src: string;
+    last_active_date: string;
+} {
```

Comment:
This might want to be a named type? Not sure if it's used enough to justify it, just noting the different from `BotInfo`.

Reply:
It's used only in two places in this file, going directly into a `render` function, so I decided it wasn't worth naming it.